ui/src/variables/components/MapVariableBuilder.tsx

```diff
@@ -0,0 +1,145 @@
+
+// Component
+// import TemplatePreviewList from 'src/tempVars/components/TemplatePreviewList'
+import {Grid, Form, TextArea, Dropdown, Columns} from '@influxdata/clockface'
+
+// Utils
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import {csvToMap, mapToCSV} from 'src/variables/utils/mapBuilder'
+
+// Actions
+import {notify as notifyAction} from 'src/shared/actions/notifications'
+
+// Constants
+import {invalidMapType} from 'src/shared/copy/notifications'
+
+type Values = {[key: string]: string}
+interface OwnProps {
+  values: Values
+  onChange: (update: {values: Values; errors: string[]}) => void
+  onSelectDefault: (selectedKey: string) => void
+  selected?: string[]
+}
+
+interface DispatchProps {
+  notify: typeof notifyAction
+}
+
+type Props = DispatchProps & OwnProps
+
+interface State {
+  templateValuesString: string
+}
+
+@ErrorHandling
+class MapVariableBuilder extends PureComponent<Props, State> {
+  state: State = {
+    templateValuesString: mapToCSV(this.props.values),
+  }
+
+  public render() {
+    const {onSelectDefault} = this.props
+    const {templateValuesString} = this.state
+    const {entries} = this
+
+    return (
+      <Form.Element label="Comma Separated Values">
+        <Grid.Row>
+          <Grid.Column>
+            <TextArea
+              value={templateValuesString}
+              onChange={this.handleChange}
+              onBlur={this.handleBlur}
+            />
+          </Grid.Column>
+        </Grid.Row>
+        <Grid.Row>
+          <Grid.Column widthXS={Columns.Six}>
+            <p>
+              Mapping Contains <strong>{entries.length}</strong> key-value pair
+              {this.pluralizer}
+            </p>
+          </Grid.Column>
+          <Grid.Column widthXS={Columns.Six}>
+            {
+              <Form.Element label="Selected Default">
+                <Dropdown
+                  selectedID={this.defaultID}
+                  onChange={onSelectDefault}
+                  titleText="Key Values"
+                >
+                  {entries.map(v => (
+                    <Dropdown.Item key={v.key} id={v.key} value={v.key}>
+                      <strong>{v.key}</strong> -> {v.value}
+                    </Dropdown.Item>
+                  ))}
+                </Dropdown>
+              </Form.Element>
+            }
+          </Grid.Column>
+        </Grid.Row>
+      </Form.Element>
+    )
+  }
+
+  private get defaultID(): string {
+    const {selected} = this.props
+    const {entries} = this
+    const firstEntry = _.get(entries, '0.key', '')
+
+    return _.get(selected, '0', firstEntry)
+  }
+
+  private get entries(): {key: string; value: string}[] {
+    const {values} = this.props
+    if (!values) {
+      return []
+    }
+
+    return Object.entries(values).map(([key, value]) => ({
+      key,
+      value,
+    }))
+  }
+
+  private get pluralizer(): string {
+    return Object.keys(this.props.values).length === 1 ? '' : 's'
```

Comment:
This should really be a utility, there are many places in the application that this is repeated.